tests/lib/rules/no-var.js

```diff
@@ -319,6 +319,74 @@ ruleTester.run("no-var", rule, {
             code: "function foo() { var { let } = {}; }",
             output: null,
             errors: [{ messageId: "unexpectedVar" }]
+        },
+
+        // https://github.com/eslint/eslint/issues/16610
+        {
+            code: "var fx = function (i = 0) { if (i < 5) { return fx(i + 1); } console.log(i); }; fx();",
```

Comment:
```suggestion
            code: "var fx = function (i = 0) { if (i < 5) { return fx(i + 1); } }; fx();",
```
In code `console.log(i)` added intentionally?

Reply:
Yes, this is the the same code (except for whitepace) as in the original issue https://github.com/eslint/eslint/issues/16610, to confirm that the reported problem is fixed.